src/cpp/lpnamer/main/ProblemGenerationExeOptions.cpp

```diff
@@ -21,16 +23,21 @@ ProblemGenerationExeOptions::ProblemGenerationExeOptions()
                            "use this option if unnamed problems are provided");
 }
 void ProblemGenerationExeOptions::Parse(unsigned int argc,
-                                        const char *const *argv) {
+                                        const char* const* argv) {
   OptionsParser::Parse(argc, argv);
   auto log_location = LOGLOCATION;
-  using namespace std::string_literals;
-  if (!XpansionOutputDir().empty() && !ArchivePath().empty()) {
-    auto msg = "Giving both archive and output options is not permitted"s;
+  if (std::vector<std::string> args = {XpansionOutputDir(), ArchivePath(),
+                                       StudyPath()};
+      std::ranges::count_if(args, std::ranges::empty) < (args.size() - 1)) {
+    auto msg = "Only one of [archive, output, study] parameters is accepted"s;
     throw ProblemGenerationOptions::ConflictingParameters(msg, log_location);
   }
-  if (XpansionOutputDir().empty() && ArchivePath().empty()) {
-    auto msg = "Both output directory and archive path are empty"s;
+  if (std::vector<std::string> mandatory = {XpansionOutputDir(), ArchivePath(),
+                                            StudyPath()};
+      std::ranges::all_of(
+          mandatory, [](std::string_view string) { return string.empty(); })) {
+    auto msg =
+        "Need to give at least on of [OutputDir, Archive, Study] options"s;
```

Comment:
bit hard to understand:
if you merge the three options with a new one, let's say --study that take a path,  the option parser will have "only" to check:
-----
if it's a zip -->archive
if not:
  if contains one of XpansionOutputdir specific file/dir:--> xpansiondir
  else: Archive
-----
may be not the smartest way though